src/app/zone-ingresses/data/index.ts

```diff
@@ -1,24 +1,100 @@
-  >>
-  & PartialZoneIngressOverview['zoneIngress']
+type PartialZoneIngressInsight = any
+type PartialInternalZoneIngress = PartialZoneIngressOverview['zoneIngress']
 
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+type InternalZoneIngress = {
+  socketAddress: string
+  advertisedSocketAddress: string
+} & Required<Pick<PartialInternalZoneIngress, 'availableServices'>> & PartialInternalZoneIngress
+
+export type ZoneIngress = {
+  config: PartialZoneIngress
+  socketAddress: string
+  advertisedSocketAddress: string
+} & Required<Pick<PartialZoneIngress, 'availableServices'>> & PartialZoneIngress
+// end TODO
+
+// TODO(jc) currently in our manually written types ZoneIngressInsight is any therefore
+// we have no Partial*
+export type ZoneIngressInsight = {
+  connectedSubscription?: KDSSubscription
+  subscriptions: KDSSubscription[]
+}
 export type ZoneIngressOverview = PartialZoneIngressOverview & {
-  zoneIngress: RequiredZoneIngress
+  zoneIngress: InternalZoneIngress
+  zoneIngressInsight?: ZoneIngressInsight
+  state: 'online' | 'offline'
+}
+// TODO(jc) Theres probably a better way to not copy/pasta this i.e. make `Entity` composable
+const InternalZoneIngress = {
+  fromObject: (item: PartialInternalZoneIngress): InternalZoneIngress => {
+    return {
+      ...item,
+      availableServices: Array.isArray(item.availableServices) ? item.availableServices : [],
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+      advertisedSocketAddress: item.networking?.advertisedAddress && item.networking?.advertisedPort ? `${item.networking.advertisedAddress}:${item.networking.advertisedPort}` : '',
+    }
+  },
+}
+
+export const ZoneIngress = {
+  fromObject: (item: PartialZoneIngress): ZoneIngress => {
+    return {
+      ...item,
+      config: item,
+      availableServices: Array.isArray(item.availableServices) ? item.availableServices : [],
+      socketAddress: item.networking?.address && item.networking?.port ? `${item.networking.address}:${item.networking.port}` : '',
+      advertisedSocketAddress: item.networking?.advertisedAddress && item.networking?.advertisedPort ? `${item.networking.advertisedAddress}:${item.networking.advertisedPort}` : '',
+    }
+  },
+}
+// end TODO
+export const ZoneIngressInsight = {
+  fromObject: (item?: PartialZoneIngressInsight): ZoneIngressInsight | undefined => {
+    // if item isn't set don't even try augmenting things
+    return isSet<PartialZoneIngressInsight>(item)
+      ? ((item) => {
+        const subscriptions: KDSSubscription[] = Array.isArray(item.subscriptions) ? item.subscriptions : []
+        // figure out the connectedSubscription by looking at the connectTime
+        // and disconnectTime of the last subscription
+        const connectedSubscription = subscriptions.slice(-1).find((item) => item.connectTime?.length && !item.disconnectTime)
```

Comment:
**Suggestion**: As discussed earlier, find the _first_ not-disconnected subscription

```suggestion
        const connectedSubscription = subscriptions.find((item) => item.connectTime?.length && !item.disconnectTime)
```

Reply:
I made an issue for this one as I think in previous PRs it still looking at the last https://github.com/kumahq/kuma-gui/issues/1890.

Seeing as we also want to amend the `connectTime` thing it would be good to do all entities/resources at once so that they are all consistent. 

If you feel strongly about it happy to do this one here now and do the others later.